### DISCUSSION TITLE
Bot Hosting Solutions

### DISCUSSION THREAD

**Discussion by lvnacy** [degen-workgroup/DEGEN#2] [Ideas]
What hosting solutions are recommended? Do you know of anything free to use? I looked at Heroku, but that looks like data costs may add up. I was planning on doing some research on the topic. If anyone has ideas for a free solution, please bring it up!

**Comment by zberwaldt** [comment]
I've been brushing up on bots with youtube. In the one video the author codes the bot on Replit then uses a service called https://uptimerobot.com/. The service pings his Replit to keep the server running.

I don't know how scalable it is, and our codebase would need to be on Replit to work. So maybe because of that, it's a no-go?

I'll keep looking for alternatives.

**Reply by lvnacy** [reply]
It’s def worth looking into. I was doing some research last night, and some hosting solutions will deploy directly from github. I run two bots on two separate Discord servers, and they’re each hosted a Digital Ocean droplet. The setup and access is easy and cheap, though we’d need to take it to a snapshot vote if we’re considering using a paid option. If we can find a solid free option, let’s do that. Otherwise, I don’t mind footing the bill for a couple bucks a month if it means we get this thing up and running sooner.

**Reply by zberwaldt** [reply]
Agreed. I also don't my chipping in for a server.

**Reply by SlinkyPotato** [reply]
Looks like discord bot development has details on implementing sharding to help with scaling https://github.com/BanklessDAO/discord-bots/discussions/2. If the bot is built right we might not have to worry to much about a scalable hosting solution.

On the contrary one can argue if Bankless DAO has a scalable server for multipurpose use, that will help with other apps and services

**Comment by pnpancholi** [comment]
I have been running a discord bot(active on many sever including bankless and DAO) on Google Cloud Run for about 2 months... not a single dollar yet. I think solutions like GCP or AWS would be great here.

**Reply by pnpancholi** [reply]
To add points to the earlier discussion, cloud run has auto-scaling and I can have min 1 container instance running to keep bot up all the time. I doubt it will cost us anything until we are using some crazy config/resources. And I doubt it will cost us anything anytime soon.

**Reply by lvnacy** [reply]
That can work. If it’s containerized, even better: we can move it to a Bankless home with ease once we figure out hosting solutions as a DAO.

**Reply by pnpancholi** [reply]
Yeah, we have to dockerize it in order to host it on Google cloud Run. It won't take me longer than 15 mins.

**Reply by lvnacy** [reply]
Awesome. I’d love to see how you get that set up. I know very little about Docker/containerization, and would love to learn all about it.

**Reply by lvnacy** [reply]
[Would this be helpful?](https://devcenter.heroku.com/categories/deploying-with-docker)

**Reply by pnpancholi** [reply]
Kind of, I can set up and push docker setup (for production) but E.O.D. Once we start writing code, I can add docker-compose so that we can develop with docker.

**Comment by lvnacy** [comment]
After doing some digging, I learned that we can deploy to Heroku and Google Cloud directly from Github, and that the bot should be fine on a free tier so long as we don’t add any music functionality to it. I think any service that we can deploy to directly from this repo will be useful.

**Reply by pnpancholi** [reply]
Yeah... we can set up Github Actions for that.

**Reply by lvnacy** [reply]
YES! That’s what I’ve been looking at. GitHub Actions will be key. Let’s do it 👍

**Reply by pnpancholi** [reply]
Cool, I have a template of sorts that I was working on. My problem there was that it wasn't detecting my GitHub Secrets.... we can work through it.

**Reply by SlinkyPotato** [reply]
Github actions is deff a plus! It can help use automate any changes that occurs esp if we set up a test guild/channel where we can all test the diff features being worked on. I created #11 so that we can track it
